### ISSUE TITLE
Regression with non-OIT renderer (unable to boot games)

### ISSUE THREAD

**Issue by diegoviola** [libretro/flycast#199]
The commit bbea5fda6fa3efbb883eeac69f6d0600f3281a62 introduced a regression when booting games with the non-OIT renderer (compiled reicast core with commented `HAVE_GL3` in the `Makefile`). Older commits like 19bbf5818e6ad46cdc94982d8a159c8729017fde do not have this issue.

```
$ ./retroarch -L ~/reicast-emulator/reicast_libretro.so ~/roms/Resident\ Evil\ -\ Code\ -\ Veronica\ \(USA\)\(Disc\ 1\ of\ 2\)\[RDC\].cdi 
[reicast]: Got size: 640 x 480.
Creating dir: /home/diego/roms/bios/dc/data
context_reset.
Failed to open /home/diego/.config/retroarch/assets/xmb/monochrome/font.ttf: No such file or directory
Failed to open /home/diego/.config/retroarch/assets/xmb/monochrome/font.ttf: No such file or directory
Linux paging: 00001000 00001000 00000FFF
MAP 00800000 w/ 25165824
MAP 20000000 w/ 25165824
MAP 04000000 w/ 16777216
MAP 06000000 w/ 16777216
MAP 0C000000 w/ 0
A8
vmem reserve: base: 83E00000, aram: a3e00000, vram: 87E00000, ram: 8FE00000
Resetting mem
Mem alloc successful!Loaded /home/diego/roms/bios/dc/dc_boot.bin as bootrom

Loaded /home/diego/roms/bios/dc/dc_nvmem.bin as nvram

Sh4 Reset
recSh4 Init
Freeing fpcb
Shader: failed to compile 
0:22(6): error: no matching function for call to `isinf(float)'; candidates are:
0:22(6): error: if-statement condition must be scalar boolean

Shader linking: failed to link 
 (51 bytes), - error: linking with uncompiled/unspecialized shader -
DEBUGBREAK!
```

My OS is Arch Linux and I'm using a Intel GPU that lacks GL 3.x support.

    00:02.0 VGA compatible controller: Intel Corporation 4 Series Chipset Integrated Graphics Controller (rev 03)

**Comment by inactive123** [comment]
@flyinghead He presented me with this issue too and I'm not entirely sure why isinf would not work for him.

Based on this link it should have been part of the GLSL spec since the very beginning -

https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/isinf.xhtml

**Comment by flyinghead** [comment]
Looks like this reference page is wrong. `isinf` was introduced in GLSL 1.30 / OpenGL 3.0.

**Comment by inactive123** [comment]
Ah, so we would have to use TARGET_GL != GL there or something? (since I added a GL3 define, that would mean it would get used for GLES2 and GL3 then.

**Comment by inactive123** [comment]
Ah wait, looks like it's not available for GLES2 either. I will use GL_TARGET == GL3 then.

**Comment by flyinghead** [comment]
Wait, I'm working on a fix already

**Comment by inactive123** [comment]
@diegoviola Check if this commit helps you -

https://github.com/libretro/reicast-emulator/commit/f00389aaf2abbed434379b7518f6bedd25dec429

**Comment by inactive123** [comment]
@flyinghead Ah sorry, jumped the gun. Let me know what you think about this commit.

**Comment by flyinghead** [comment]
It's a tad bit more complicated ;)

**Comment by inactive123** [comment]
Ah OK. I will let you lead on this issue then.

**Comment by flyinghead** [comment]
This PR #200 should take care of the problem

**Comment by inactive123** [comment]
Alright, merged.

**Comment by diegoviola** [comment]
I can confirm that the bug is fixed with the current master branch (commit 19defa865ba812d569fbaee29e737f11a990bbcf).

Thanks for fixing this issue. :)
